### ISSUE TITLE
Spec development process

### ISSUE THREAD

**Issue by constantinpape** [ome/ngff#45]
I am a bit confused by the different versions of the spec that are stored in this repository:
We have the specs for the individual versions, right now `0.1` and `0.2` and we have `latest`.
Currently, `0.2` is further ahead then `latest`, because it contains the changes to the chunk layout from #40, which are not in `latest` yet.

In the future, I would propose a more structured dev process:
All changes are made against `latest` and only when we draft a new release, we copy `latest` to the folder for the corresponding version and add this with a separate commit (which only copies the spec and does not introduce any further changes).

cc @joshmoore

**Comment by joshmoore** [comment]
Agreed, @constantinpape. This was my mistake because I was thinking you "owned" latest while I was making a non-breaking change. Something that hadn't been done before. The changes that just got merged to `0.2` will need backporting to `latest`. I can take care of that if you are not already in the middle of it.

P.S. It might be though that these static copies aren't ideal anyway and we want something less prone to conflicts.

**Comment by constantinpape** [comment]
> Agreed, @constantinpape. This was my mistake because I was thinking you "owned" latest while I was making a non-breaking change. Something that hadn't been done before.

Sure, no worries. Luckily the changes are still very manageable, so it's not problem right now to sync this manually.

> The changes that just got merged to `0.2` will need backporting to `latest`. I can take care of that if you are not already in the middle of it.

I am in the middle of it already, see #46. I just need to sync the latest review changes later.

> P.S. It might be though that these static copies aren't ideal anyway and we want something less prone to conflicts.

I think the static copies are fine, if we keep development strictly limited to `latest`.

**Comment by constantinpape** [comment]
Ok, after looking at the review comments in #46: I guess it would be best to introduce
some placeholder for the version number in `latest` and then write a script that makes the copy and replaces the placeholder with the correct version.

**Comment by joshmoore** [comment]
Hmm... on call, but I don't see an immedaite way to replace a variable. Files can be included if that would help: https://tabatkins.github.io/bikeshed/#including

**Comment by constantinpape** [comment]
> Hmm... on call, but I don't see an immedaite way to replace a variable.

Ok, I will have a closer look at bikeshed too and see if I can find something.
If there is nothing in there, we could also raise an issue about it.
Alternatively, it would be fairly easy to write some script for this ourselves (but we would need to introduce our own bikeshed flavor)

> Files can be included if that would help: https://tabatkins.github.io/bikeshed/#including

Thanks for the pointer.

**Comment by joshmoore** [comment]
Not to mention the need for this! https://github.com/joshmoore/spec-prod/commit/3fa5978babcbf9ccb9db491710465075c18caf65

**Comment by sbesson** [comment]
I assume this will be covered by the post-release item mentioned in https://github.com/ome/ngff/issues/86

**Comment by constantinpape** [comment]
> I assume this will be covered by the post-release item mentioned in #86

Yes, that's the plan ;).

**Comment by lubianat** [comment]
Hi,

Thank you all for the discussion here. As part of an effort to keep the NGFF issue tracker manageable, we're closing tickets that haven't seen activity in several years and whose discussion may be out of date.

If this issue is still relevant, feel free to reopen it with details in the current context of NGFF.

Best,
Tiago
